logger.go

```diff
@@ -3,195 +3,274 @@ package ntest
+	// Get caller information, walking up the stack to find the first non-helper function
+	var file string
+	var line int
+
+	// Get multiple frames at once and walk through them
+	const maxFrames = 32
+	pcs := make([]uintptr, maxFrames)
+	n := runtime.Callers(3, pcs) // Skip logMessage + Log/Logf + caller
+	frames := runtime.CallersFrames(pcs[:n])
+
+	for {
+		frame, more := frames.Next()
+
+		// Skip internal logger functions and marked helpers
+		if !bl.isHelper(frame.Function) {
+			file = frame.File
+			line = frame.Line
 			// Get just the filename, not the full path
-			if idx := strings.LastIndex(file, "/"); idx >= 0 {
-				file = file[idx+1:]
-			}
+			file = filepath.Base(file)
+			break
 		}
 
-		entry := bufferedLogEntry{
-			message: message,
-			file:    file,
-			line:    line,
+		if !more {
+			file = "unknown"
+			line = 0
+			break
 		}
+	}
 
-		lock.Lock()
-		defer lock.Unlock()
+	entry := bufferedLogEntry{
+		message: message,
+		file:    file,
+		line:    line,
+	}
 
-		if cleanupCalled {
-			t.Logf("[%s:%d] %s", file, line, message)
-		} else {
-			entries = append(entries, entry)
-		}
+	bl.entryLock.Lock()
+	defer bl.entryLock.Unlock()
+
+	if bl.cleanupCalled {
+		bl.T.Helper()
+		bl.T.Logf("[%s:%d] %s", file, line, message)
+	} else {
+		bl.entries = append(bl.entries, entry)
 	}
 }
 
-// BufferedLogger creates a logger wrapper that buffers all log output and only
-// outputs it during test cleanup if the test failed. Each log entry
-// includes the filename and line number where the log was called.
-// The purpose of this is for situations where go tests are defaulting
-// to -v but output should be suppressed anyway.
-//
-// If the environment variable NTEST_BUFFERING is set to "false", buffering
-// will be turned off and the original T will be returned directly.
-//
-// One advantage of using BufferedLogger over using "go test" (without -v) is
-// that you can see the skipped tests with BufferedLogger whereas non-v go test
-// hides the skips.
-func BufferedLogger[ET T](t ET) T {
-	if os.Getenv("NTEST_BUFFERING") == "false" {
-		// When buffering is disabled, return the original T directly to avoid any intermediate calls
-		return t
-	}
+// Log method for bufferedLoggerT that uses the buffered logger function
+func (bl *bufferedLoggerT[ET]) Log(args ...interface{}) {
+	bl.Helper() // Call our own Helper method to track this as a helper
+	line := fmt.Sprintln(args...)
+	message := line[0 : len(line)-1]
+	bl.logMessage(message)
+}
+
+// Logf method for bufferedLoggerT that uses the buffered logger function
+func (bl *bufferedLoggerT[ET]) Logf(format string, args ...interface{}) {
+	bl.Helper() // Call our own Helper method to track this as a helper
+	message := fmt.Sprintf(format, args...)
+	bl.logMessage(message)
+}
+
+// Helper method for bufferedLoggerT that tracks helpers
+func (bl *bufferedLoggerT[ET]) Helper() {
+	// Mark the caller of Helper as a helper
+	bl.markHelper(0)
+	MarkHelpers(bl.T)
+}
```

Comment:
Should this also call `bl.T.Helper()`?

Reply:
No. There is no point in calling an inner .Helper because the assumed count of stack frames will be off.

Hmm. I just had an idea: what if .Helper automatically skipped any parent function named .Helper?  Then I could just call inner .Helper and wouldn't need the MarkHelpers interface.

And I could suggest the same fix for testing.T